Scripts/XRTrailRenderer.cs

```diff
@@ -20,131 +22,84 @@ public class VRTrailRenderer : MonoBehaviour
     bool m_StealLastPointWhenEmpty = true;
 
     [SerializeField]
-    Color[] m_Colors;
+    [Tooltip("How long the tail should be (second) [ 0, infinity ].")]
+    float m_Time = 5.0f;
+
+    [SerializeField]
+    [Tooltip("The minimum distance to spawn a new point on the trail [ 0, infinity ].")]
+    float m_MinVertexDistance = 0.1f;
+
+    [SerializeField]
+    [Tooltip("Destroy GameObject when there is no trail?")]
+    bool m_Autodestruct = false;
+
+    [SerializeField]
+    [Tooltip("With this enabled, the last point will smooth lerp between the last recorded anchor point and the one after it")]
+    bool m_SmoothInterpolation = false; 
 
     // Circular array support for trail point recording
     Vector3[] m_Points;
     float[] m_PointTimes;
     int m_PointIndexStart = 0;
     int m_PointIndexEnd = 0;
 
-
     // Cached Data
-    VRMeshChain m_VrMeshData;
-    bool m_MeshNeedsRefreshing = false;
-    Renderer m_MeshRenderer;
     Vector3 m_LastRecordedPoint = Vector3.zero;
 
     int m_UsedPoints;			// How many points we are currently rendering - for size/color blending
     float m_LastPointTime;
 
     float m_EditorDeltaHelper;  // This lets us have access to a time data while not in play mode
 
-    [SerializeField]
-    float m_Time = 5.0f;
 
-    // TrailRenderer Interface
+    /// <summary>
+    /// How long does the trail take to fade out.
+    /// </summary>
     public float time
     {
-        get 
-        { 
-            return m_Time; 
-        }
-        set 
-        { 
-            m_Time = value; 
-        }
-    }
-
-    [SerializeField]
-    float m_StartWidth = 1.0f;
-
-    public float startWidth
-    {
-        get
-        {
-            return m_StartWidth;
-        }
-        set
-        {
-            m_StartWidth = value;
-        }
-    }
-
-    [SerializeField]
-    float m_EndWidth = 1.0f;
-
-    public float endWidth
-    {
-        get
-        {
-            return m_EndWidth;
-        }
-        set
-        {
-            m_EndWidth = value;
-        }
+        get { return m_Time; }
+        set { m_Time = value; }
```

Comment:
another place we should maybe clamp setting of the value >= 0
it doesn't break if you set a negative time value, but it doesn't make sense either